demos/draw-hooks.html

```diff
@@ -0,0 +1,237 @@
+<!doctype html>
+<html>
+	<head>
+		<meta charset="utf-8">
+		<title>Custom Paint</title>
+		<meta name="viewport" content="width=device-width, initial-scale=1">
+
+		<link rel="stylesheet" href="../bench/style.css">
+		<link rel="stylesheet" href="../src/uPlot.css">
+		<style>
+			.uplot {
+				margin-top: 50px;
+			}
+		</style>
+	</head>
+	<body>
+		<script src="../dist/uPlot.iife.js"></script>
+		<script>
+
+            function addCrazyVisualsPlugin() {
+
+                // Source: https://stackoverflow.com/questions/21646738/convert-hex-to-rgba
+                function hexToRgbA(hex, a='1') {
+                    let c;
+                    if(/^#([A-Fa-f0-9]{3}){1,2}$/.test(hex)){
+                        c= hex.substring(1).split('');
+                        if(c.length== 3){
+                            c= [c[0], c[0], c[1], c[1], c[2], c[2]];
+                        }
+                        c= '0x'+c.join('');
+                        return 'rgba('+[(c>>16)&255, (c>>8)&255, c&255].join(',')+',' + a +')';
+                    }
+                    throw new Error('Bad Hex');
+                }
+
+                // Source: https://stackoverflow.com/questions/25837158/how-to-draw-a-star-by-using-canvas-html5
+                function drawStar(ctx, cx,cy,spikes=6,outerRadius=8,innerRadius=4){
+                    var rot=Math.PI/2*3;
+                    var x=cx;
+                    var y=cy;
+                    var step=Math.PI/spikes;
+
+                    ctx.beginPath();
+                    ctx.moveTo(cx,cy-outerRadius)
+                    for(i=0;i<spikes;i++){
+                        x=cx+Math.cos(rot)*outerRadius;
+                        y=cy+Math.sin(rot)*outerRadius;
+                        ctx.lineTo(x,y)
+                        rot+=step
+
+                        x=cx+Math.cos(rot)*innerRadius;
+                        y=cy+Math.sin(rot)*innerRadius;
+                        ctx.lineTo(x,y)
+                        rot+=step
+                    }
+                    ctx.lineTo(cx,cy-outerRadius);
+                    ctx.closePath();                    
+                }
+
+                function setCtxProperties(ctx, settings)
```

Comment:
i'm not a canvas expert, but isn't this effectively https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/save?